discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -127,11 +130,44 @@ def _get_domain_ips(self, domain):
         ips, _ = self._get_domain_ips_and_macs(domain)
         return ips
 
-    def _wait_till_domain_has_ips(self, domain, timeout=120, interval=5):
+    def _wait_till_domain_has_ips(self, domain, timeout=240, interval=5):
         logging.info("Waiting till host %s will have ips", domain.name())
         waiting.wait(
             lambda: len(self._get_domain_ips(domain)) > 0,
             timeout_seconds=timeout,
             sleep_seconds=interval,
             waiting_for="Waiting for Ips",
         )
+
+    def set_boot_order(self, node_name, cd_first=False):
+        logging.info(f"Going to set the following boot order: cd_first: {cd_first}, "
+                     f"for node: {node_name}")
+        command = f"virsh dumpxml {node_name}"
+        current_xml = utils.run_command(command)
```

Comment:
maybe use libvirt client to get the xml? somthing like
raw_xml = self.libvirt_connection.lookupByName()
xml = minidom.parseString(raw_xml)

reference: https://www.libvirt.org/docs/libvirt-appdev-guide-python/en-US/html/libvirt_application_development_guide_using_python-Guest_Domains-Device_Config-Disks.html